modules/op-custom-parameter-expansion.adoc

```diff
@@ -0,0 +1,69 @@
+// This module is included in the following assembly:
+//
+// *cicd/pipelines/using-pipelines-as-code.adoc
+
+:_content-type: CONCEPT
+[id="op-custom-parameter-expansion_{context}"]
+= Custom parameter expansion
+
+{pac} helps you expanding a custom parameter within your `PipelineRun` resource by using the `params` field. You can specify a value for the custom parameter inside the template of the `Repository` CR. The specified value replaces the custom parameter in your pipeline run.
+
+[NOTE]
+====
+Use the custom parameter expansion feature only in specific scenarios when you cannot use the Tekton `PipelineRun` parameters.
```

Comment:
Here is some material to add @mramendi : 

It's generally better to use Tekton params instead of custom paac parameters, because tekton parameters can have logics which the pipelinerun understand. The tekton parameters are part of the pipelinerun and due versioned with the pipelinerun.

The paac parameters is something defined on the Repository CR, the values are defined in context of the payload received. They are simply key/value that will be applied on the pipelinerun before getting applied on cluster.

One typical use case would be defining a parameters for a URL (like a registry URL) that will change depending if it's a push or a pull_request.

Another use case is a parameter defined by administrators (like a account UUID) which can be used by users on the pipelinerun. The parameter can be managed by the administrator without having to make a change to the user pipelinerun in the git repository.

Reply:
@chmouel could you please provide an example how a Tekton parameter can "incorporate logic that the pipeline comprehends"?

Also, you write "The paac parameters, on the other hand, are defined within the Repository CR and their values are determined based on the received payload." However, the Tekton parameters are defined in the pipeline and the pipeline is defined in the payload, so they are determined based on the payload too. What is the difference here?

"A typical scenario for using Paac parameters is when defining a URL parameter (such as a registry URL) that varies depending on whether it's a push or a pull_request." - I don't get this too. You specify the value for the parameter in the Repository CR, so how does it become different for push vs pull_request?

Reply:
your guesswork is correct, the paac parameters defined on Repository CR are not versioned with the code in the .tekton/ folder

Tekton Parameters is a core functionality of a PipelineRun, https://tekton.dev/docs/pipelines/pipelineruns/#specifying-parameters as you see from the documentation they do much more than PaaC/Repository CR Parameters which is only a key/value replacement.

here is an example of a PaaC repo params which sets value based on the context payload:

```yaml
spec:
  params:
    - name: company
      value: "My Beautiful Company"
      filter:
        - name: event
          value: |
            pac.event_type == "pull_request"
```

the advantage compared to Tekton Params is highlighted in our upstream [documentation](https://pipelinesascode.com/docs/guide/repositorycrd/#cel-filtering-on-custom-parameters):

```
You can have multiple params with the same name and different filters, the first param that matches the filter will be picked up. This let you have different output according to different event, and for example combine a push and a pull request event.
```